scripts/edit-task.js

```diff
@@ -0,0 +1,230 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+import createButton from "/scripts/factory/createButton.js";
+import toggleButtons from "/scripts/factory/toggleButton.js";
+
+function saveEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  tasks[id].content = inputElement.textContent;
+  toggleButtons(
+    editButton,
+    deleteButton,
+    revertButton,
+    saveButton,
+    doneButton,
+    saveAndDoneButton
+  );
+  saveButton.removeEventListener(CLICK_EVENT, saveEventHandler);
+  revertButton.removeEventListener(CLICK_EVENT, revertEventHandler);
+  saveAndDoneButton.removeEventListener(CLICK_EVENT, saveAndDoneEventHandler);
+}
+
+function onTodoDone(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  toolbar,
+  tasks,
+  id,
+  editEventHandler,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  inputElement.classList.add("done");
+  const task = tasks[id];
+  task.done = true;
+  task.content = inputElement.textContent;
+
+  editButton.removeEventListener(CLICK_EVENT, editEventHandler);
+  // doneButton.removeEventListener(CLICK_EVENT, onTodoDone);
+
+  toggleButtons(saveAndDoneButton, saveButton, revertButton, deleteButton);
+
+  const diffDays = getTaskCompletedDays(task.date.getTime());
+  const completedText =
+    diffDays === 1 ? `Completed in 1 day` : `Completed in ${diffDays} days`;
+  const completeTimeBtn = createButton(
+    "complete-time",
+    completedText,
+    "Time to complete the task"
+  );
+  task.completeTime = diffDays;
+  toolbar.append(completeTimeBtn);
+
+  saveButton.removeEventListener(CLICK_EVENT, saveEventHandler);
+  revertButton.removeEventListener(CLICK_EVENT, revertEventHandler);
+  saveAndDoneButton.removeEventListener(CLICK_EVENT, saveAndDoneEventHandler);
+}
+
+function revertEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  prevContent,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.innerText = prevContent;
+  inputElement.contentEditable = false;
+  tasks[id].content = inputElement.textContent;
```

Comment:
redundant DOM access.

```suggestion
  inputElement.innerText = prevContent;
  inputElement.contentEditable = false;
  tasks[id].content = prevContent;
```